### PR TITLE
*: add the tid to the schema-change related tables only when keys need to be locked

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve?

Issue Number: close #xxx 
https://github.com/pingcap/tidb/issues/15565

Problem Summary:
In https://github.com/pingcap/tidb/pull/3999, TiDB always adds the tid to the schema change-related tables no matter there're keys really need to be locked. 

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:
Add them to the schema change-related tables just when there're some keys that need to be locked.

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Breaking backward compatibility

### Release note 
Add the table ID to the schema-change related tables only when keys need to be locked in select for update statement.

### CODE REVIEW

File: session/session_test.go [hunk 1]
```diff
@@ -2099,6 +2099,15 @@ func (s *testSchemaSuite) TestRetrySchemaChangeForEmptyChange(c *C) {
 	tk.MustExec("create table t1 (i int)")
 	tk.MustExec("begin")
 	tk1.MustExec("alter table t add j int")
+	tk.MustExec("select * from t for update")
+	tk.MustExec("update t set i = -i")
+	tk.MustExec("delete from t")
+	tk.MustExec("insert into t1 values (1)")
+	tk.MustExec("commit")
```
Comment (jackysp) [inline session/session_test.go]: This test will fail on release-3.0. So I marked this PR as a compatibility breaker.
\cc @zhouqiang-cl @you06

**Comment by cfzjywxk** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**